### ISSUE TITLE
Allow package name in git tag

### ISSUE THREAD

**Issue by jsamr** [release-it/release-it#708]
I attempted to configure `tagName` to `"${name}@${version}"`, but release-it complained that "ERROR ReferenceError: name is not defined", whilst this pattern is working for `github.releaseName` and `git.commitMessage`. I believe it would be more consistent to make the name available in `git.tagName`. Especially useful for monorepos, to follow a pattern [such as in react-navigation library](https://github.com/react-navigation/react-navigation/releases).

**Comment by webpro** [comment]
Thanks @jsamr, I've just add the full context in `git.tagName` in [v14.0.4](https://github.com/release-it/release-it/releases/tag/14.0.4). Let me know how it works for you :)

**Comment by jsamr** [comment]
@webpro After playing a little with this, I think for this to work properly, we would also need to change how latest tag is assessed. Currently, [this is done with](https://github.com/release-it/release-it/blob/2ccf01fd242309840020cfa82034ade4851c6cf7/lib/plugin/GitBase.js#L91):

```js
"git describe --tags --abbrev=0"
```

What I would need is:

```js
`git describe --tags --match "${tagName.replace(version, "*")}" --abbrev=0`
```

Also, I believe [`getLatestVersion`](https://github.com/release-it/release-it/blob/2ccf01fd242309840020cfa82034ade4851c6cf7/lib/plugin/GitBase.js#L27) should be changed accordingly.

**Comment by jsamr** [comment]
@webpro I could implement a quick and dirty fix like so:

```diff
diff --git a/lib/plugin/GitBase.js b/lib/plugin/GitBase.js
index dcf35ed4cafb14e1f823a3929e5534039e690652..3697a76d78549b4e4710c797fabdb49ef3499bac 100644
--- a/lib/plugin/GitBase.js
+++ b/lib/plugin/GitBase.js
@@ -14,7 +14,7 @@ class GitBase extends Plugin {
     await this.fetch();
     const repo = parseGitUrl(this.remoteUrl);
     const latestTagName = await this.getLatestTagName();
-    const secondLatestTagName = this.options.isUpdate ? await this.getSecondLatestTagName() : null;
+    const secondLatestTagName = this.options.isUpdate ? await this.getSecondLatestTagName(latestTagName) : null;
     const tagTemplate = this.options.tagName || ((latestTagName || '').match(/^v/) ? 'v${version}' : '${version}');
     this.setContext({ repo, tagTemplate, latestTagName, secondLatestTagName });
     this.config.setContext({ latestTag: latestTagName });
@@ -89,14 +89,17 @@ class GitBase extends Plugin {
   }
 
   getLatestTagName() {
-    return this.exec('git describe --tags --abbrev=0', { options }).then(
+    const match = format(this.options.tagName, { version: "*" })
+    const gitCmd = `git describe --tags --match "${match}" --abbrev=0`;
+    return this.exec(gitCmd, { options }).then(
       stdout => stdout || null,
       () => null
     );
   }
 
-  async getSecondLatestTagName() {
-    const sha = await this.exec('git rev-list --tags --skip=1 --max-count=1', { options });
+  async getSecondLatestTagName(latestTag) {
+    const match = format(this.options.tagName, { version: "*" })
+    const sha = await this.exec(`git rev-list ${latestTag} --tags --max-count=1`, { options });
     return this.exec(`git describe --tags --abbrev=0 ${sha}`, { options }).catch(() => null);
   }
 }
```
However, I'm dumbfounded that `@release-it/conventional-changelog` still picks the wrong previous tag, generating a CHANGELOG grabbing all changes across different workspaces. I add also patch the former with a little addition to select only commits which alter the path encompassing a workspace, like so:

```diff
diff --git a/index.js b/index.js
index abbce1815895b044bac47588a327f1b049304b8d..e1266a84c7984ee6f4944e0c4595c7410a69bed0 100644
--- a/index.js
+++ b/index.js
@@ -68,7 +68,8 @@ class ConventionalChangelog extends Plugin {
       Object.assign(options, this.options),
       { version, previousTag, currentTag },
       {
-        debug: this.config.isDebug ? this.debug : null
+        debug: this.config.isDebug ? this.debug : null,
+        path: './'
       }
     );
   }
``` 
So the only thing that is breaking in my monorepo right now, is the fact the logic to determine `latestTag` is not shared across plugins...

**Comment by jsamr** [comment]
I managed to get `@release-it/conventional-changelog` work with my workspace setup thanks to the following changes:

``` diff
diff --git a/index.js b/index.js
index abbce1815895b044bac47588a327f1b049304b8d..e31f4ee93c6896fefb30552e8bf89f6c182610a4 100644
--- a/index.js
+++ b/index.js
@@ -1,6 +1,7 @@
 const { EOL } = require('os');
 const fs = require('fs');
 const { Plugin } = require('release-it');
+const { format } = require('release-it/lib/util');
 const conventionalRecommendedBump = require('conventional-recommended-bump');
 const conventionalChangelog = require('conventional-changelog');
 const semver = require('semver');
@@ -14,7 +15,8 @@ class ConventionalChangelog extends Plugin {
 
   getInitialOptions(options, namespace) {
     options[namespace].tagName = options.git.tagName;
-    return options[namespace];
+    const tagPrefix = format(options.git.tagName || '${version}', { version: '' });
+    return {...options[namespace], path: './', tagPrefix};
   }
 
   async getChangelog(latestVersion) {
@@ -68,7 +70,8 @@ class ConventionalChangelog extends Plugin {
       Object.assign(options, this.options),
       { version, previousTag, currentTag },
       {
-        debug: this.config.isDebug ? this.debug : null
+        debug: this.config.isDebug ? this.debug : null,
+        path: this.options.path
       }
     );
   }
```

**Comment by jsamr** [comment]
@webpro Would you rather that I file an issue for workspaces and enumerate the required changes? Wouldn't mind crafting a PR if you are open to the idea.

**Comment by webpro** [comment]
Thanks @jsamr, I'm definitely open to such improvements. Just didn't have the time to dig into this, so a PR would help!

**Comment by webpro** [comment]
@SuaYoo's work is released in `release-it@next` ([v14.3.0-next.0](https://github.com/release-it/release-it/releases/tag/14.3.0-next.0)).If anybody has a chance, please let me know if everything is working as expected.
